content/en/docs/advanced/creating_custom_cloud_images.md

```diff
@@ -0,0 +1,151 @@
+
+## Prerequisites
+
+Before starting, ensure you have:
+
+- Docker installed and running
+- A Linux machine with KVM support (for testing)
+- Basic understanding of container images and cloud configurations
+
+## Step 1: Creating a Base Image with kairos-init
+
+The first step is to create a base container image using [`kairos-init`](https://github.com/kairos-io/kairos-init). This tool allows you to create a custom base image from popular distributions.
+
+Here's a basic example of creating a base image:
+
+```bash
+docker build -t my-custom-image - <<EOF
+ARG BASE_IMAGE=ubuntu:24.04
+
+FROM quay.io/kairos/kairos-init:v0.4.3 AS kairos-init
+
+FROM ${BASE_IMAGE} AS base-kairos
+ARG VARIANT=core
+ARG MODEL=generic
+ARG TRUSTED_BOOT=false
+ARG KUBERNETES_DISTRO=k3s
+ARG KUBERNETES_VERSION=latest
+ARG FRAMEWORK_VERSION=v2.20.0
+ARG VERSION
+
+COPY --from=kairos-init /kairos-init /kairos-init
+RUN /kairos-init -f "${FRAMEWORK_VERSION}" -l debug -s install -m "${MODEL}" -v "${VARIANT}" -t "${TRUSTED_BOOT}" -k "${KUBERNETES_DISTRO}" --k8sversion "${KUBERNETES_VERSION}" --version "${VERSION}"
+RUN /kairos-init -f "${FRAMEWORK_VERSION}" -l debug -s init -m "${MODEL}" -v "${VARIANT}" -t "${TRUSTED_BOOT}" -k "${KUBERNETES_DISTRO}" --k8sversion "${KUBERNETES_VERSION}" --version "${VERSION}"
+RUN /kairos-init -f "${FRAMEWORK_VERSION}" -l debug --validate -m "${MODEL}" -v "${VARIANT}" -t "${TRUSTED_BOOT}" -k "${KUBERNETES_DISTRO}" --k8sversion "${KUBERNETES_VERSION}" --version "${VERSION}"
+RUN rm /kairos-init
+EOF
+```
+
+## Step 2: Customizing the Image
+
+Once you have a base image, you can customize it by:
+
+1. Adding packages
+2. Modifying configurations
+3. Adding custom services
+4. Including additional files
+
+See the [Customizing Images]({{< ref "/docs/advanced/customizing" >}}) guide for detailed instructions.
+
+## Step 3: Building Bootable Images with AuroraBoot
+
+After creating and customizing your base image, you can use AuroraBoot to create bootable images:
+
+```bash
+docker run -v "$PWD"/build:/tmp/auroraboot \
+             -v /var/run/docker.sock:/var/run/docker.sock \
+             --rm -ti quay.io/kairos/auroraboot \
+             --set container_image=my-custom-image \
+             --set "disable_http_server=true" \
+             --set "disable_netboot=true" \
+             --cloud-config /path/to/cloud-config.yaml \
+             --set "state_dir=/tmp/auroraboot"
+```
+
+{{% alert title="Note" color="info" %}}
+For more details about AuroraBoot options and configurations, see the [AuroraBoot documentation]({{< ref "/docs/reference/auroraboot" >}}).
+{{% /alert %}}
+
+### State Partition Sizing
+
+When creating cloud images, it's important to consider the size of the state partition. The state partition is created at image build time and cannot be resized later. This partition needs to accommodate all images (passive, active, and recovery) that might be used with the system.
```

Comment:
state only stores active+passive and a transition image for upgrades, recovery is not in this partition